tasks/pipeline.py

```diff
@@ -991,3 +996,72 @@ def test_merge_queue(ctx):
     ctx.run(f"git push origin :{test_main}", hide=True)
     if not success:
         raise Exit(message="Merge queue test failed", code=1)
+
+
+@task
+def compare_to_itself(ctx):
+    """
+    Create a new branch with 'compare_to_itself' in gitlab-ci.yml and trigger a pipeline
+    """
+    if not gitlab_configuration_is_modified(ctx):
+        print("No modification in the gitlab configuration, ignoring this test.")
+        return
+    agent = get_gitlab_repo()
+    gh = GithubAPI()
+    current_branch = os.getenv("CI_COMMIT_REF_NAME")
+    new_branch = f"compare/{current_branch}/{int(datetime.now().timestamp())}"
+    ctx.run(f"git checkout -b {new_branch}", hide=True)
+    ctx.run(
+        f"git remote set-url origin https://x-access-token:{gh._auth.token}@github.com/DataDog/datadog-agent.git",
+        hide=True,
+    )
+    ctx.run("git config --global user.name 'github-app[bot]'")
+    ctx.run("git config --global user.email 'github-app[bot]@users.noreply.github.com'")
+    # The branch must exist in gitlab to be able to "compare_to"
+    ctx.run(f"git push origin {new_branch}", hide=True)
+    time.sleep(15)  #  Wait for the branch to be created
+    for file in ['.gitlab-ci.yml', '.gitlab/notify/notify.yml']:
+        with open(file) as f:
+            content = f.read()
+        with open(file, 'w') as f:
+            f.write(content.replace('compare_to: main', f'compare_to: {new_branch}'))
+    ctx.run("git commit -am 'Compare to itself'", hide=True)
+    ctx.run(f"git push origin {new_branch}", hide=True)
+    max_attempts = 6
+    for attempt in range(max_attempts):
+        print(f"[{datetime.now()}] Waiting 30s for the pipelines to be created")
+        time.sleep(30)
+        pipelines = agent.pipelines.list(per_page=20, get_all=False)
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**🥜 nitpick:** Iirc we can list by ref using `ref=...`, this would be faster and more explicit‏

Reply:
That's a nice one indeed, thanks